content/fr/docs/tasks/configure-pod-container/configure-pod-configmap.md

```diff
@@ -0,0 +1,693 @@
+
+```shell
+kubectl apply -k .
+configmap/game-config-4-m9dm2f92bt created
+```
+
+Vous pouvez vérifier que le ConfigMap a été créé comme ceci:
+
+```text
+kubectl get configmap
+NAME                       DATA   AGE
+game-config-4-m9dm2f92bt   1      37s
+
+
+kubectl describe configmaps/game-config-4-m9dm2f92bt
+Name:         game-config-4-m9dm2f92bt
+Namespace:    default
+Labels:       <none>
+Annotations:  kubectl.kubernetes.io/last-applied-configuration:
+                {"apiVersion":"v1","data":{"game.properties":"enemies=aliens\nlives=3\nenemies.cheat=true\nenemies.cheat.level=noGoodRotten\nsecret.code.p...
+
+Data
+====
+game.properties:
+----
+enemies=aliens
+lives=3
+enemies.cheat=true
+enemies.cheat.level=noGoodRotten
+secret.code.passphrase=UUDDLRLRBABAS
+secret.code.allowed=true
+secret.code.lives=30
+Events:  <none>
+```
+
+Notez que le nom ConfigMap généré a un suffixe obtenu par hachage de son contenu.
+Cela garantit qu'un nouveau ConfigMap est généré chaque fois que le contenu est modifié.
+
+#### Définissez la clé à utiliser lors de la génération d'un ConfigMap à partir d'un fichier
+
+Vous pouvez définir une clé autre que le nom de fichier à utiliser dans le générateur ConfigMap.
+Par exemple, pour générer un ConfigMap à partir du fichier `configure-pod-container/configmap/game.properties`
+avec la clé `game-special-key`
+
+```shell
+# Créer un fichier kustomization.yaml avec ConfigMapGenerator
+cat <<EOF >./kustomization.yaml
+configMapGenerator:
+- name: game-config-5
+  files:
+  - game-special-key=configure-pod-container/configmap/game.properties
+EOF
+```
+
+Appliquer le dossier kustomization pour créer l'objet ConfigMap.
+
+```text
+kubectl apply -k .
+configmap/game-config-5-m67dt67794 created
+```
+
+#### Générer des ConfigMaps à partir de littéraux
+
+Pour générer un ConfigMap à partir de littéraux `special.type=charm` et `special.how=very`, vous pouvez spécifier le générateur ConfigMap dans `kustomization.yaml` comme
+
+```shell
+# Create a kustomization.yaml file with ConfigMapGenerator
+cat <<EOF >./kustomization.yaml
+configMapGenerator:
+- name: special-config-2
+  literals:
+  - special.how=very
+  - special.type=charm
+EOF
+```
+
+Appliquez le dossier kustomization pour créer l'objet ConfigMap.
+
+```text
+kubectl apply -k .
+configmap/special-config-2-c92b5mmcf2 created
+```
+
+## Définir des variables d'environnement de conteneur à l'aide des données ConfigMap
+
+### Définissez une variable d'environnement de conteneur avec les données d'une seule ConfigMap
+
+1. Définissez une variable d'environnement comme paire clé-valeur dans un ConfigMap:
+
+    ```shell
+    kubectl create configmap special-config --from-literal=special.how=very
+    ```
+
+1. Attribuez la valeur `special.how` défini dans ConfigMap à la variable d'environnement `SPECIAL_LEVEL_KEY` dans la spécification du Pod.
+
+   {{< codenew file="pods/pod-single-configmap-env-variable.yaml" >}}
```

Comment:
the path is specific to the localization, you must copy the file to the fr branch,
/opt/build/repo/content/fr/examples/pods/pod-single-configmap-env-variable.yaml: no such file or directory